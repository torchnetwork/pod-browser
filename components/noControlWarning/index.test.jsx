/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { renderWithTheme } from "../../__testUtils/withTheme";
import NoControlWarning, {
  setupOnClose,
  TESTCAFE_ID_NO_CONTROL_ERROR,
} from "./index";

afterEach(() => sessionStorage.clear());

describe("NoControlWarning", () => {
  it("renders", () => {
    const { asFragment, queryByTestId } = renderWithTheme(
      <NoControlWarning podRootIri="http://example.com" />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId(TESTCAFE_ID_NO_CONTROL_ERROR)).toBeTruthy();
  });

  it("renders nothing if no-control-warning-closed is set to true", () => {
    sessionStorage.setItem("no-control-warning-closed", "true");
    const { asFragment, queryByTestId } = renderWithTheme(
      <NoControlWarning podRootIri="http://example.com" />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId(TESTCAFE_ID_NO_CONTROL_ERROR)).toBeFalsy();
  });
});

describe("setupOnClose", () => {
  it("allows user to close message", () => {
    expect(sessionStorage.getItem("no-control-warning-closed")).toBeNull();
    const event = { preventDefault: jest.fn() };
    expect(setupOnClose()(event)).toBeUndefined();
    expect(event.preventDefault).toHaveBeenCalledWith();
    expect(sessionStorage.getItem("no-control-warning-closed")).toEqual("true");
  });
});
