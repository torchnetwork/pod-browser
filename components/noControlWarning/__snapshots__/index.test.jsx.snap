// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoControlWarning renders 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-message PodBrowser-message--warning PodBrowser-message--prominent"
    data-testid="no-control-error"
  >
    <div
      class="PodBrowser-message__icon PodBrowser-message__icon--warning"
    >
      <span
        class="PodBrowser-icon-error"
      />
    </div>
    <div
      class="PodBrowser-message__main"
    >
      <p
        class="PodBrowser-message__title"
      >
        You can't share anything or change permissions in your Pod
      </p>
      <p>
        This is because you don't have control access. You can still browse the resources in your Pod, but you won't be able to share anything.
      </p>
      <p
        class="PodBrowser-message__title"
      >
        How to fix:
      </p>
      <ol>
        <li>
          <span>
            Log in to 
          </span>
          <a
            href="http://example.com"
            rel="noreferrer"
            target="_blank"
          >
            <span>
              your Pod
            </span>
            <span
              class="PodBrowser-icon-external-link"
              data-external="true"
            />
          </a>
        </li>
        <li>
          Go to “Preferences”
        </li>
        <li>
          <span>
            Under “Manage your trusted applications,” look for 
          </span>
          <span>
            http://localhost
          </span>
          <span>
             and check the “Control” box
          </span>
        </li>
        <li>
          Click on “Update”
        </li>
        <li>
          Return to PodBrowser
        </li>
      </ol>
    </div>
    <div
      class="PodBrowser-message__close PodBrowser-message__close--warning"
    >
      <button
        class="PodBrowser-button PodBrowser-button--text"
        data-testid="no-control-close-button"
      >
        <span
          aria-label="Close message"
          class="PodBrowser-icon-cancel"
          role="img"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NoControlWarning renders nothing if no-control-warning-closed is set to true 1`] = `<DocumentFragment />`;
